core/src/main/java/com/sequenceiq/cloudbreak/conclusion/step/Conclusion.java

```diff
@@ -0,0 +1,53 @@
+package com.sequenceiq.cloudbreak.conclusion.step;
+
+public class Conclusion {
+
+    private final Class<? extends ConclusionStep> conclusionStepClass;
+
+    private final boolean stepFailed;
+
+    private final String conclusion;
+
+    private final String details;
+
+    public Conclusion(boolean stepFailed, String conclusion, String details, Class<? extends ConclusionStep> conclusionStepClass) {
+        this.stepFailed = stepFailed;
+        this.conclusion = conclusion;
+        this.details = details;
+        this.conclusionStepClass = conclusionStepClass;
+    }
+
+    public static Conclusion succeeded(Class<? extends ConclusionStep> conclusionStepClass) {
+        return new Conclusion(false, null, null, conclusionStepClass);
+    }
+
+    public static Conclusion failed(String conclusion, String details, Class<? extends ConclusionStep> conclusionStepClass) {
+        return new Conclusion(true, conclusion, details, conclusionStepClass);
+    }
+
+    public boolean isStepFailed() {
```

Comment:
This StepFailed notation is somewhat ambiguous to me. 
Does it detected a failure before it could finish or is there execution error?

Might be better with 2 different values? 
```suggestion
    public boolean isStepFailed() {
```
IsCompleted() and isFailureFound()

Reply:
The conclusion object is created only when the execution is finished. IsCompleted always would be true.